hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/HoodieSparkRecordMerger.java

```diff
@@ -70,9 +71,11 @@ public Option<Pair<HoodieRecord, Schema>> merge(HoodieRecord older, Schema oldSc
       }
     }
     if (older.getOrderingValue(oldSchema, props).compareTo(newer.getOrderingValue(newSchema, props)) > 0) {
-      return Option.of(Pair.of(older, oldSchema));
+      return Option.of(SparkPartialMergingUtils.mergePartialRecords(
+          (HoodieSparkRecord) newer, newSchema, (HoodieSparkRecord) older, oldSchema, props));
```

Comment:
The partial merge may not happen, so maybe give the utility a better name.

Reply:
Makes sense.  Renamed to `SparkRecordMergingUtils#mergeCompleteOrPartialRecords`